core/payment/src/models/agreement.rs

```diff
@@ -39,7 +38,7 @@ impl WriteObj {
         let payment_platform = demand_properties
             .pointer("/golem/com/payment/chosen-platform")
             .as_typed(Value::as_str)
-            .unwrap_or(DEFAULT_PAYMENT_PLATFORM)
+            .expect("Demand property golem.com.payment.chosen-platform does not exist")
```

Comment:
As before, it panics, so it should be either converted to Error or removed if unnecessary (because of new Negotiator).